SequenceAnalysis/resources/web/SequenceAnalysis/panel/SequenceImportPanel.js

```diff
@@ -1392,13 +1393,209 @@ Ext4.define('SequenceAnalysis.panel.SequenceImportPanel', {
+                                    }
+
+                                    var rows = [];
+                                    Ext4.Array.forEach(s, function(record){
+                                        var recId = record.get('fileRecord1');
+                                        var data = this.fileNameStore.snapshot || this.fileNameStore.data;
+                                        var fileDataRecIdx = data.findIndexBy(function(r){
+                                            return r.get('id') === recId;
+                                        });
+                                        var fileDataRec = data.getAt(fileDataRecIdx);
+                                        if (!fileDataRec) {
+                                            return;
+                                        }
+
+                                        var val = fileDataRec.get('fileName');
+                                        var row = [val, null, null, record];
+                                        if (groupEx) {
+                                            var m = val.match(groupEx);
+                                            if (m && m.length) {
+                                                if (groupNum && groupNum < m.length) {
+                                                    row[1] = m[groupNum];
+                                                }
+                                            }
+                                        }
+
+                                        if (platformEx) {
+                                            var m = val.match(platformEx);
+                                            if (m && m.length) {
+                                                if (platformNum && platformNum < m.length) {
+                                                    row[2] = m[platformNum];
+                                                }
+                                            }
+                                        }
+
+                                        rows.push(row);
+                                    }, this);
+
+                                    rows.sort(function(a, b){
+                                        if (a[1] > b[1]) return 1;
+                                        if (b[1] > a[1]) return -1;
+
+                                        return 0;
+                                    });
+
+                                    var dataItems = [{
+                                        html: 'File'
+                                    },{
+                                        html: 'Updated Group'
+                                    },{
+                                        html: 'Updated Platform Unit'
+                                    }];
+
+                                    Ext4.Array.forEach(rows, function(r){
+                                        dataItems.push({
+                                            html: r[0]
```

Comment:
Does this need to be HTML-encoded or is it always safe?

Reply:
I though Ext4 did this when it renders the grid (the code that uses this), but I will check.  It's probably quite low risk, but a good reminder